src/frontCANnode/src/input.c

```diff
@@ -0,0 +1,178 @@
+#include "input.h"
+#include "adc.h"
+
+#define ADC_UPDATE_PERIOD_MS 10
+
+#define ACCEL_1_LOWER_BOUND 255
+#define ACCEL_1_UPPER_BOUND 799
+#define ACCEL_2_LOWER_BOUND 257
+#define ACCEL_2_UPPER_BOUND 677
+
+// Some wheel speed stuff (copied from MY17)
+#define WHEEL_SPEED_TIMEOUT_MS 100
+#define WHEEL_SPEED_READ_PERIOD_MS 10
+
+// Microsecond = 1 millionth of a second
+#define MICROSECONDS_PER_SECOND_F 1000000.0
+// 1000 millirevs = 1 rev
+#define MILLIREVS_PER_REV_F 1000.0
+#define SECONDS_PER_MINUTE 60
+
+void update_adc(Input_T *input);
+void update_wheel_speed(Speed_Input_T *speed, uint32_t msTicks);
+void update_can(Input_T *input);
+
+uint16_t transform1(uint32_t accel_1_raw);
+uint16_t transform2(uint32_t accel_2_raw);
+
+
+void Input_initialize(Input_T *input) {
+  input->adc->accel_1 = 0;
+  input->adc->accel_2 = 0;
+  input->adc->accel_1_raw = 0;
+  input->adc->accel_2_raw = 0;
+  input->adc->brake_1 = 0;
+  input->adc->brake_2 = 0;
+  input->adc->steering_pot = 0;
+}
+
+void Input_fill_input(Input_T *input) {
+  update_adc(input);
+  update_wheel_speed(input->speed, input->msTicks);
+}
+
+void update_adc(Input_T *input) {
+  ADC_Input_T *adc = input->adc;
+  uint32_t next_updated = adc->last_updated + ADC_UPDATE_PERIOD_MS;
```

Comment:
Nice function.
Consider doing `static uint32_t last_updated = 0;` and then saving last_updated there instead of in the struct.